src/gui/app.cpp

```diff
@@ -348,12 +445,29 @@ void App::showVersionInfo(QCommandLineParser& parser) noexcept
 	PrintInfo(versionInfo);
 }
 
-void App::mainWindowClosing(int status) {
-	m_exitStatus = status;
-}
+void App::openNewWindow(const QVariantList& args) noexcept
+{
+	QString server;
+	QString nvim{ "nvim" };
+	ConnectorInitArgs::Type type{ ConnectorInitArgs::Type::Default };
+	if (args.size() > 1) {
+		Q_ASSERT(args.at(1).type() == QVariant::Type::Map);
+		auto initMap = args.at(1).toMap();
+		if (initMap.find("nvim") != initMap.end()) {
+			nvim = initMap["nvim"].toString();
+		}
+
+		if (initMap.find("server") != initMap.end()) {
+			server = initMap["server"].toString();
+			type = ConnectorInitArgs::Type::Server;
+		}
+	}
 
-void App::exitWithStatus() {
-	exit(m_exitStatus);
+	auto connector = connectToRemoteNeovim(
```

Comment:
nitpick:

Avoid `auto` if the type is simple. `auto` is good for iterators, complex templates.

`auto` is bad for simple types, the type has to be inferred by the reader. Fine for compiler, hard for humans.